src/services/point.service.ts

```diff
@@ -1,22 +1,78 @@
-import { Guild } from 'discord.js';
+import { Guild, Role, TextChannel, MessageEmbed, EmojiIdentifierResolvable } from 'discord.js';
 import { IUserPoints, Maybe } from '../common/types';
 import { PointsDocument, PointsModel } from '../schemas/points.schema';
 import { GuildService } from './guild.service';
+import Constants from '../common/constants';
+import { UserService } from './user.service';
+import ms from 'ms';
 
 export class PointService {
   private _guild: Guild;
+  private _userservice: UserService;
+  private _tacoKingRole: Role;
+  private _tacoKingEmoji: Maybe<EmojiIdentifierResolvable> = null;
 
-  constructor(private _guildService: GuildService) {
+  constructor(private _guildService: GuildService, private _userService: UserService) {
     this._guild = this._guildService.get();
+    this._userservice = this._userService;
+    this._tacoKingRole = this._guildService.getRole(Constants.Roles.TacoKing);
+    this._tacoKingEmoji = this._guildService.getEmoji('tacoking') ?? '🌮';
   }
 
   public async awardPoints(id: Maybe<string>, amount: number) {
     if (!id) {
       return;
     }
 
+    // cache the #1 point holder before points are awarded
+    const [{ userID: kingBeforeAwardID }] = await this.getTopPoints(1);
+
     const userDoc = await this.getUserPointDoc(id);
     await PointsModel.updateOne(userDoc, { $inc: { numPoints: amount } });
+
+    const [{ userID: kingAfterAwardID }] = await this.getTopPoints(1);
+
+    // the king has changed!
+    if (kingBeforeAwardID !== kingAfterAwardID) {
+      // get the old and new king members
+      const [prevKingUser, newKingUser] = [kingBeforeAwardID, kingAfterAwardID].map((c) =>
+        this._userservice.getMember(c)
+      );
+
+      // remove the taco king role from the previous king
+      // and give the role to the new king
+      await Promise.all([
+        prevKingUser?.roles.remove(this._tacoKingRole),
```

Comment:
Did you test what happens if prevKingUser is undefined or null here?  Does it still resolve to an empty promise